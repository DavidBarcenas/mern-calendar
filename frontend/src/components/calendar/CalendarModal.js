import React, { useEffect, useState } from 'react'
import { Button, Dialog, DialogActions, DialogContent, DialogTitle, Icon, IconButton, Slide, TextField } from '@material-ui/core';
import { useDispatch, useSelector } from 'react-redux';
import { addNewEvent, deleteEvent, eventSetActive, updateEvent } from '../../redux/actions/events';
import { uiCloseModal } from '../../redux/actions/ui';
import { dateFormat } from '../../utils/format';
import moment from 'moment'

const Transition = React.forwardRef(function Transition(props, ref) {
  return <Slide direction="up" ref={ ref } { ...props } />;
});

const initEvent = {
  title: '',
  notes: '',
  start: '',
  end:   ''
}

export const CalendarModal = () => {

  const dispatch = useDispatch()
  const {modalOpen} = useSelector(state => state.ui)
  const {activeEvent} = useSelector(state => state.calendar)
  const [formValues, setFormValues] = useState(initEvent)
  const [inputError, setInputError] = useState({
    startError: false,
    endError:   false,
    titleError: false
  })

  const { title, notes, start, end } = formValues
  const { startError, endError, titleError } = inputError

  useEffect(() => {
    if(activeEvent) {
      setFormValues({
        ...activeEvent,
        start: dateFormat(activeEvent.start),
        end: dateFormat(activeEvent.end),
      })
    } else {
      setFormValues(initEvent)
    }
  }, [activeEvent])

  const handleInputChange = ({target}) => {
    setFormValues({
      ...formValues,
      [target.name]: target.value
    })
  }

  const validateForm = () => {
    let startDay = moment(start)
    let endDate = moment(end)    

    setInputError({ 
      startError: !startDay.isValid(),
      endError:   !endDate.isValid(),
    })

    if (!startDay.isValid() || !endDate.isValid()) {
      return false
    }

    if (title.trim().length <= 2) {
      setInputError({
        ...formValues,
        titleError: true,
      })
      return false;
    }

    if (startDay.isSameOrAfter(endDate)) {
      setInputError({
        ...formValues,
        endError: true,
      })
      return false;
    }

    setInputError({ 
      startError: false,
      endError:   false,
      titleError: false
    })

    return true;
  }

  const handleSubmit = () => {
    if(validateForm()) {
      if(activeEvent && !activeEvent.slot) {
        dispatch(updateEvent({
          ...formValues,
          start: moment(start).toDate(),
          end: moment(end).toDate(),
        }))
      } else {
        dispatch(addNewEvent({ 
          ...formValues, 
          start: moment(start).toDate(),
          end: moment(end).toDate(),
          id: new Date().getTime() ,
          user: {
            _id: 'abc123',
            name: 'David Barcenas'
          }
        }))
      }

      closeModal()
    }
  }

  const closeModal = () => {
    dispatch( uiCloseModal() )
    dispatch( eventSetActive(null) )
  }

  const handleDelete = () => {
    dispatch(deleteEvent())
    dispatch(uiCloseModal())
  }

  return (
    <Dialog
      aria-labelledby="alert-dialog-slide-title"
      aria-describedby="alert-dialog-slide-description"
      className="calendarModal"
      open={modalOpen}
      TransitionComponent={Transition}
      keepMounted
    >
      <DialogTitle id="alert-dialog-slide-title">
        <span>{activeEvent && !activeEvent.slot ? 'Editar evento' : 'Crear evento'}</span>
        <IconButton onClick={closeModal}> <Icon>Cancel</Icon> </IconButton >
      </DialogTitle>
      <DialogContent>
        <form noValidate autoComplete="off">
          <TextField
            onChange={handleInputChange}
            onBlur={validateForm}
            error={startError}
            id="datetime-local"
            label="Fecha y hora de inicio"
            type="datetime-local"
            name="start"
            helperText={startError ? 'Este campo es obligatorio' : ''}
            value={start}
            InputLabelProps={{
              shrink: true,
            }}
            
          />
          <TextField
            onChange={handleInputChange}
            onBlur={validateForm}
            error={endError}
            id="datetime-local"
            label="Fecha y hora final"
            type="datetime-local"
            name="end"
            helperText={endError ? 'Este campo es obligatorio y no puede ser menor a la fecha de inicio' : ''}
            value={end}
            InputLabelProps={{
              shrink: true
            }}
          />
          <TextField 
            id="outlined-basic" 
            label="Titulo" 
            name="title" 
            variant="outlined" 
            value={title}
            error={titleError}
            helperText={titleError ? 'Este campo es obligatorio' : ''}
            onChange={handleInputChange} 
            onBlur={validateForm}
          />
          <TextField
            id="outlined-multiline-static"
            label="Notas"
            multiline
            rows={4}
            variant="outlined"
            name="notes"
            value={notes}
            onChange={handleInputChange} 
          />
        </form>
      </DialogContent>
      <DialogActions>
        {
          activeEvent && !activeEvent.slot  ? 
            <Button onClick={handleDelete} color="secondary"> Eliminar </Button> :
            <Button onClick={closeModal} color="primary"> Cancelar </Button>
        }
        <Button variant="contained" type="button" color="primary" className="btn-primary" onClick={handleSubmit}> Guardar </Button>
      </DialogActions>
    </Dialog>
  )
}
