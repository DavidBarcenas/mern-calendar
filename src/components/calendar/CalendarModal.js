import React, { useState } from 'react'
import moment from 'moment'
import { Button, Dialog, DialogActions, DialogContent, DialogTitle, Slide, TextField } from '@material-ui/core';

const Transition = React.forwardRef(function Transition(props, ref) {
  return <Slide direction="up" ref={ref} {...props} />;
});

export const CalendarModal = ({ event, open, setOpen  }) => {

  const [inputError, setInputError] = useState({
    startError: false,
    endError:   false,
    titleError: false
  })

  const [formValues, setFormValues] = useState({
    title: 'Evento',
    notes: '',
    start: '',
    end:   ''
  })

  const { title, notes, start, end } = formValues
  const { startError, endError, titleError } = inputError

  const handleInputChange = ({ target }) => {
    setFormValues({
      ...formValues,
      [target.name]: target.value
    })

    
  }

  const validateForm = () => {
    let startDay = moment(start)
    let endDate = moment(end)    

    setInputError({ 
      startError: !startDay.isValid(),
      endError:   !endDate.isValid(),
    })
    
    if (!startDay.isValid() || !endDate.isValid()) {
      return false
    }

    if (title.trim().length <= 2) {
      setInputError({
        ...formValues,
        titleError: true,
      })
      return false;
    }

    if (startDay.isSameOrAfter( endDate )) {
      setInputError({
        ...formValues,
        endError: true,
      })
      return false;
    }

    setInputError({ 
      startError: false,
      endError:   false,
      titleError: false
    })

    return true;
  }

  const handleSubmit = () => {
    console.log('form', validateForm())

    if( validateForm() ) {
      console.log('el value', formValues)
    }
  }

  return (
    <Dialog
      open={true}
      TransitionComponent={Transition}
      keepMounted
      onClose={() => setOpen(false)}
      aria-labelledby="alert-dialog-slide-title"
      aria-describedby="alert-dialog-slide-description"
      className="calendarModal"
    >
      <DialogTitle id="alert-dialog-slide-title">Crear un evento</DialogTitle>

      <DialogContent>
      <form noValidate autoComplete="off">
        <TextField
          onChange={ handleInputChange }
          error={ startError }
          id="datetime-local"
          label="Fecha y hora de inicio"
          type="datetime-local"
          name="start"
          helperText={ startError ? 'Este campo es obligatorio' : ''}
          InputLabelProps={{
            shrink: true,
          }}
        />
        <TextField
          onChange={ handleInputChange }
          error={ endError }
          id="datetime-local"
          label="Fecha y hora final"
          type="datetime-local"
          name="end"
          helperText={ endError ? 'Este campo es obligatorio y no puede ser menor a la fecha de inicio' : ''}
          InputLabelProps={{
            shrink: true
          }}
        />

        <TextField 
          id="outlined-basic" 
          label="Titulo" 
          name="title" 
          variant="outlined" 
          value={ title }
          error={ titleError }
          helperText={ titleError ? 'Este campo es obligatorio' : ''}
          onChange={ handleInputChange } 
        />

        <TextField
          id="outlined-multiline-static"
          label="Notas"
          multiline
          rows={4}
          variant="outlined"
          name="notes"
          value={ notes }
          onChange={ handleInputChange } 
        />
      </form>
      </DialogContent>
      <DialogActions>
        <Button onClick={() => setOpen(false)} color="primary">
          Cancelar
        </Button>
        <Button variant="contained" type="button" color="primary" className="btn-primary" onClick={ handleSubmit }>
          Guardar
        </Button>
      </DialogActions>
    </Dialog>
  )
}
