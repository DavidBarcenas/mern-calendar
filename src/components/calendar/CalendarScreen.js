import React, { useState } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { Calendar, momentLocalizer } from 'react-big-calendar'
import { Navbar } from '../ui/Navbar'
import { messages_es } from '../../utils/messages-es'
import { CalendarEvent } from './CalendarEvent'
import { CalendarModal } from './CalendarModal'
import { uiOpenModal } from '../../redux/actions/ui'
import { eventSetActive } from '../../redux/actions/events'
import { Fab, Icon } from '@material-ui/core'
import { eventFormat } from '../../utils/format'
import moment from 'moment'

import 'moment/locale/es-mx'
import 'react-big-calendar/lib/css/react-big-calendar.css'

moment.locale('es-mx')
const localizer = momentLocalizer(moment)
const formats = { 
  timeGutterFormat: 'h a', 
  eventTimeRangeFormat: ({ start, end }) =>
  moment(start).format('h:mma') + ' - ' + moment(end).format('h:mma')
}

export const CalendarScreen = () => {

  const dispatch = useDispatch()
  const { events, activeEvent } = useSelector(state => state.calendar)
  const [lastView, setLastView] = useState( localStorage.getItem('lastView') || 'month' )
  
  const onDoubleClick = (e) => {
    dispatch( uiOpenModal() )
  }
  
  const onSelectEvent = (e) => {
    dispatch(eventSetActive(e))
  }

  const onViewChange = (e) => {
    setLastView(e)
  }

  const handleSelectSlot = (e) => {
    let event = eventFormat(e)
    if( e.action === 'doubleClick' && !activeEvent ) {
      event.slot = true
      dispatch(eventSetActive(event))
      dispatch( uiOpenModal() )
    }
  }

  const btnAddNew = () => {
    dispatch( uiOpenModal() )
  }

  const eventStyleGetter = (event, start, end, isSelected) => {
    const style = {
      backgroundColor: '#396afc',
      borderLeft: '3px solid #396afc',
      borderRadius: '0',
      color: '#ffffff',
      display: 'block',
      fontSize: '14px',
      opacity: 0.8,
    }

    return { style }
  }
  
  return (
    <div className="calendar__wrapper">
      <Navbar />

      <div>
        <Calendar
          formats={ formats }
          localizer={ localizer }
          events={ events }
          startAccessor="start"
          endAccessor="end"
          messages={ messages_es }
          eventPropGetter={ eventStyleGetter }
          onDoubleClickEvent={ onDoubleClick }
          onSelectEvent={ onSelectEvent }
          onView={ onViewChange }
          onSelectSlot={ handleSelectSlot }
          selectable={ true }
          view={ lastView }
          components={{
            event: CalendarEvent
          }}
        />
      </div>

      <CalendarModal />

      <Fab color="primary" aria-label="add" className="btn-new" onClick={ btnAddNew }>
        <Icon>add</Icon>
      </Fab>
    </div>
  )
}
