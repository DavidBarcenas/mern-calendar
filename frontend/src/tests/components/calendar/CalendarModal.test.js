import '@testing-library/jest-dom';
import { mount } from 'enzyme';
import React from 'react'
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk';
import { CalendarModal } from '../../../components/calendar/CalendarModal';

const middlewares = [thunk]
const mockStore = configureStore(middlewares)

const initState = {
  ui: {
    modalOpen: false
  }, 
  calendar: {
    activeEvent: {
      title:"Prueba de evento",
      notes:"Una nota del evento",
      start:"2020-10-09T05:00:00.000Z",
      end:"2020-10-09T06:00:00.000Z",
      id:"5f7f93f93fa7870588f89610",
      user: {
        _id:"5f7be3a8014f613100b4bd71",
        name:"David"
      }
    }
  }
}
const store = mockStore(initState)
store.dispatch = jest.fn()

const wrapper = mount(
  <Provider store={store}>
    <CalendarModal />
  </Provider>
)

describe('<CalendarModal /> testing', () => {
  test('should render conrrectly', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
