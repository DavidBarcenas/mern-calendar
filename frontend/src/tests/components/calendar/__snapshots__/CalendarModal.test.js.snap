// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CalendarModal /> testing should render conrrectly 1`] = `
<Portal
  containerInfo={
    <body>
      <div
        aria-hidden="true"
        class="MuiDialog-root calendarModal"
        role="presentation"
        style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px; visibility: hidden;"
      >
        <div
          aria-hidden="true"
          class="MuiBackdrop-root"
          style="opacity: 0; visibility: hidden;"
        />
        <div
          data-test="sentinelStart"
          tabindex="0"
        />
        <div
          class="MuiDialog-container MuiDialog-scrollPaper"
          role="none presentation"
          style="visibility: hidden; webkit-transform: translateY(768px) translateY(0px); transform: translateY(768px) translateY(0px);"
          tabindex="-1"
        >
          <div
            aria-describedby="alert-dialog-slide-description"
            aria-labelledby="alert-dialog-slide-title"
            class="MuiPaper-root MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiPaper-elevation24 MuiPaper-rounded"
            role="dialog"
          >
            <div
              class="MuiDialogTitle-root"
              id="alert-dialog-slide-title"
            >
              <h2
                class="MuiTypography-root MuiTypography-h6"
              >
                <span>
                  Editar evento
                </span>
                <button
                  class="MuiButtonBase-root MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                     
                    <span
                      aria-hidden="true"
                      class="material-icons MuiIcon-root"
                    >
                      cancel
                    </span>
                     
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </h2>
            </div>
            <div
              class="MuiDialogContent-root"
            >
              <form
                autocomplete="off"
                novalidate=""
              >
                <div
                  class="MuiFormControl-root MuiTextField-root"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
                    data-shrink="true"
                    for="datetime-local"
                    id="datetime-local-label"
                  >
                    Fecha y hora de inicio
                  </label>
                  <div
                    class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiInput-input"
                      id="datetime-local"
                      name="start"
                      type="datetime-local"
                      value="2020-10-09T00:00:00"
                    />
                  </div>
                  
                </div>
                <div
                  class="MuiFormControl-root MuiTextField-root"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
                    data-shrink="true"
                    for="datetime-local"
                    id="datetime-local-label"
                  >
                    Fecha y hora final
                  </label>
                  <div
                    class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiInput-input"
                      id="datetime-local"
                      name="end"
                      type="datetime-local"
                      value="2020-10-09T01:00:00"
                    />
                  </div>
                  
                </div>
                <div
                  class="MuiFormControl-root MuiTextField-root"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
                    data-shrink="true"
                    for="outlined-basic"
                    id="outlined-basic-label"
                  >
                    Titulo
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input"
                      id="outlined-basic"
                      name="title"
                      type="text"
                      value="Prueba de evento"
                    />
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="PrivateNotchedOutline-legendLabelled-3 PrivateNotchedOutline-legendNotched-4"
                      >
                        <span>
                          Titulo
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                  
                </div>
                <div
                  class="MuiFormControl-root MuiTextField-root"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
                    data-shrink="true"
                    for="outlined-multiline-static"
                    id="outlined-multiline-static-label"
                  >
                    Notas
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
                  >
                    <textarea
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
                      id="outlined-multiline-static"
                      name="notes"
                      rows="4"
                    >
                      Una nota del evento
                    </textarea>
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="PrivateNotchedOutline-legendLabelled-3 PrivateNotchedOutline-legendNotched-4"
                      >
                        <span>
                          Notas
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </form>
            </div>
            <div
              class="MuiDialogActions-root MuiDialogActions-spacing"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                   Eliminar 
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained btn-primary MuiButton-containedPrimary"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                   Guardar 
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
        <div
          data-test="sentinelEnd"
          tabindex="0"
        />
      </div>
    </body>
  }
>
  <div
    className="MuiDialog-root calendarModal"
    onKeyDown={[Function]}
    role="presentation"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "fixed",
        "right": 0,
        "top": 0,
        "visibility": "hidden",
        "zIndex": 1300,
      }
    }
  >
    <div
      aria-hidden={true}
      className="MuiBackdrop-root"
      onClick={[Function]}
      style={
        Object {
          "opacity": 0,
          "visibility": "hidden",
        }
      }
    />
    Array [
      <div
        data-test="sentinelStart"
        tabIndex={0}
      />,
      <div
        className="MuiDialog-container MuiDialog-scrollPaper"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        role="none presentation"
        style={
          Object {
            "visibility": "hidden",
          }
        }
        tabIndex="-1"
      >
        <div
          aria-describedby="alert-dialog-slide-description"
          aria-labelledby="alert-dialog-slide-title"
          className="MuiPaper-root MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiPaper-elevation24 MuiPaper-rounded"
          role="dialog"
        >
          <div
            className="MuiDialogTitle-root"
            id="alert-dialog-slide-title"
          >
            <h2
              className="MuiTypography-root MuiTypography-h6"
            >
              <span>
                Editar evento
              </span>
              <button
                className="MuiButtonBase-root MuiIconButton-root"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiIconButton-label"
                >
                   
                  <span
                    aria-hidden={true}
                    className="material-icons MuiIcon-root"
                  >
                    cancel
                  </span>
                   
                </span>
                <span
                  className="MuiTouchRipple-root"
                />
              </button>
            </h2>
          </div>
          <div
            className="MuiDialogContent-root"
          >
            <form
              autoComplete="off"
              noValidate={true}
            >
              <div
                className="MuiFormControl-root MuiTextField-root"
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
                  data-shrink={true}
                  htmlFor="datetime-local"
                  id="datetime-local-label"
                >
                  Fecha y hora de inicio
                </label>
                <div
                  className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    autoFocus={false}
                    className="MuiInputBase-input MuiInput-input"
                    disabled={false}
                    id="datetime-local"
                    name="start"
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={false}
                    type="datetime-local"
                    value="2020-10-09T00:00:00"
                  />
                </div>
              </div>
              <div
                className="MuiFormControl-root MuiTextField-root"
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
                  data-shrink={true}
                  htmlFor="datetime-local"
                  id="datetime-local-label"
                >
                  Fecha y hora final
                </label>
                <div
                  className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    autoFocus={false}
                    className="MuiInputBase-input MuiInput-input"
                    disabled={false}
                    id="datetime-local"
                    name="end"
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={false}
                    type="datetime-local"
                    value="2020-10-09T01:00:00"
                  />
                </div>
              </div>
              <div
                className="MuiFormControl-root MuiTextField-root"
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
                  data-shrink={true}
                  htmlFor="outlined-basic"
                  id="outlined-basic-label"
                >
                  Titulo
                </label>
                <div
                  className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    autoFocus={false}
                    className="MuiInputBase-input MuiOutlinedInput-input"
                    disabled={false}
                    id="outlined-basic"
                    name="title"
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={false}
                    type="text"
                    value="Prueba de evento"
                  />
                  <fieldset
                    aria-hidden={true}
                    className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      className="PrivateNotchedOutline-legendLabelled-3 PrivateNotchedOutline-legendNotched-4"
                    >
                      <span>
                        Titulo
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <div
                className="MuiFormControl-root MuiTextField-root"
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
                  data-shrink={true}
                  htmlFor="outlined-multiline-static"
                  id="outlined-multiline-static-label"
                >
                  Notas
                </label>
                <div
                  className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
                  onClick={[Function]}
                >
                  <textarea
                    aria-invalid={false}
                    autoFocus={false}
                    className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
                    disabled={false}
                    id="outlined-multiline-static"
                    name="notes"
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={false}
                    rows={4}
                    value="Una nota del evento"
                  />
                  <fieldset
                    aria-hidden={true}
                    className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      className="PrivateNotchedOutline-legendLabelled-3 PrivateNotchedOutline-legendNotched-4"
                    >
                      <span>
                        Notas
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </form>
          </div>
          <div
            className="MuiDialogActions-root MuiDialogActions-spacing"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                 Eliminar 
              </span>
              <span
                className="MuiTouchRipple-root"
              />
            </button>
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained btn-primary MuiButton-containedPrimary"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                 Guardar 
              </span>
              <span
                className="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>,
      <div
        data-test="sentinelEnd"
        tabIndex={0}
      />,
    ]
  </div>
</Portal>
`;
